core/vm/logger.go

```diff
@@ -60,6 +62,50 @@ type StructLog struct {
 	Depth   int
 	Err     error
 }
+type StructLogJsonOut struct {
+	Pc      uint64              `json:"pc"`
+	Op      OpCode              `json:"op"`
+	OpName  string              `json:"opName"`
+	Gas     math.HexOrDecimal64 `json:"gas"`
+	GasCost math.HexOrDecimal64 `json:"gasCost"`
+	Memory  string              `json:"memory"`
+	Stack   hexArray            `json:"stack"`
```

Comment:
You can use gencodec to avoid writing the conversion.

```go
//go:generate gencodec -type StructLogJson -field-override structLogJsonMarshaling -out gen_structlog_json.go

type StructLogJson struct{
    ...
    Stack []*big.Int
}

type structLogJsonMarshaling struct{
    Stack []*math.HexOrDecimal256
}
```

This would create a MarshalJSON method that converts Stack internally.